homeassistant/components/camera/__init__.py

```diff
@@ -945,6 +1016,95 @@ async def ws_camera_web_rtc_offer(
         connection.send_result(msg["id"], {"answer": answer})
 
 
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "camera/web_rtc_config",
+        vol.Required("entity_id"): cv.entity_id,
+    }
+)
+@websocket_api.async_response
+async def ws_camera_web_rtc_config(
+    hass: HomeAssistant, connection: ActiveConnection, msg: dict[str, Any]
+) -> None:
+    """Get configuration for the web rtc session such as ice servers.
+
+    Async friendly.
+    """
+    entity_id = msg["entity_id"]
+    camera = _get_camera_from_entity_id(hass, entity_id)
+    if camera.frontend_stream_type != StreamType.WEB_RTC:
+        connection.send_error(
+            msg["id"],
+            "web_rtc_config_failed",
+            (
+                "Camera does not support WebRTC,"
+                f" frontend_stream_type={camera.frontend_stream_type}"
+            ),
+        )
+        return
```

Comment:
Move this to a decorator to be used here and in  `ws_camera_web_rtc_close` to avoid the duplication